File: Prolog/eu002.pl
:- initialization(search).

search :-
    search(0, 1, 0, Sum),
    print(Sum), nl, halt.

search(_, X2, Acc, Acc) :- X2 > 4000000.
search(X1, X2, Acc, Result) :-
    X3 is X1 + X2,
    sum_even(X3, Acc, NewAcc),
    search(X2, X3, NewAcc, Result).

sum_even(X, Acc, NewAcc) :-
    0 is mod(X, 2),
    NewAcc is Acc + X.
sum_even(_, Acc, Acc).

File: C/eu002.c
#include <stdio.h>

int main() {
    int x1 = 0, x2 = 1, x3, sum = 0;

    while (x2 < 4000000) {
        if (x2 % 2 == 0) {
            sum += x2;
        }
        x3 = x1 + x2;
        x1 = x2;
        x2 = x3;
    }

    printf("%d\n", sum);
}
